### ISSUE TITLE
Dependency based-extractors

### ISSUE THREAD

**Issue by danyaljj** [IllinoisCogComp/illinois-cogcomp-nlp#106]
I came across this when using a bunch of dependency-based feature extractors: 

Consider for example [`DependencyPathNgrams`](https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/blob/master/edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/DependencyPathNgrams.java#L86-Lundefined). If I mistakenly give a constituent which is **not** a dependency view, the feature extractor would fail, since [a few lines later](https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/blob/master/edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/DependencyPathNgrams.java#L93) we do `c.getIncomingRelations().get(0)`. Instead I think the code should be changed from: 
```java 
  @Override
    public Set<Feature> getFeatures(Constituent c) throws EdisonException {
        TextAnnotation ta = c.getTextAnnotation();

        TreeView parse = (TreeView) ta.getView(dependencyViewName);

        Constituent c1 =
                parse.getConstituentsCoveringToken(
                        c.getIncomingRelations().get(0).getSource().getStartSpan()).get(0);
        Constituent c2 = parse.getConstituentsCoveringToken(c.getStartSpan()).get(0);
```
to 

```java 
  @Override
    public Set<Feature> getFeatures(Constituent c) throws EdisonException {
        TextAnnotation ta = c.getTextAnnotation();

        TreeView parse = (TreeView) ta.getView(dependencyViewName);

        // Reordering: first find equivalent of c in the parse view 
        Constituent c2 = parse.getConstituentsCoveringToken(c.getStartSpan()).get(0);

       // main change is here: c changed to c2 
        Constituent c1 =
                parse.getConstituentsCoveringToken(
                        c2.getIncomingRelations().get(0).getSource().getStartSpan()).get(0);
```

what do you think?
I have seen this issue in at least 4-5 other extractors as well. Other places with similar issue: 
- [SyntacticFrame](https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/blob/master/edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/SyntacticFrame.java#L49-Lundefined)

- [ParsePath](https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/blob/master/edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ParsePath.java#L68-L69)

- [ClauseFeatureExtractor](https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/blob/master/edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java) 

- ChunkPathPattern

**Comment by christos-c** [comment]
I don't get why this is better than what we have. If `c2` doesn't have the an incoming relation this would also fail. Also, why isn't this failing during unit tests?

**Comment by danyaljj** [comment]
I think that's a separate issue; even if `c` is in a dependency view, it might not have an incoming relation (which can cause the code to fail with dependency view, as it did for me). 

My point is, for a fresh user it is not clear that the input `Constituent c` should be from a dependency view. 
But since we already know the dependency view name inside `dependencyViewName`, why don't we always manually lookup the equivalent of `Constituent c` form `dependencyViewName`, if it is already from that view?

**Comment by christos-c** [comment]
Oh I see, yes ok that makes sense. Go ahead and make the change everywhere.

**Comment by christos-c** [comment]
Closed by #109
